spec/controllers/v1.0/actions_controller_spec.rb

```diff
@@ -59,4 +58,40 @@
       end
     end
   end
+
+  describe 'POST /requests/:request_id/actions' do
+    context 'when request is actionable' do
+      let(:req) { create(:request, :with_context, :tenant_id => tenant.id) }
+      let!(:stage1) { create(:stage, :id => "1", :state => Stage::NOTIFIED_STATE, :request => req, :tenant_id => tenant.id) }
+      let!(:stage2) { create(:stage, :id => "2", :state => Stage::PENDING_STATE, :request => req, :tenant_id => tenant.id) }
+      let(:valid_attributes) { { :operation => 'cancel', :processed_by => 'abcd' } }
+
+      before do
+        allow(Group).to receive(:find)
+        post "#{api_version}/requests/#{req.id}/actions", :params => valid_attributes, :headers => request_header
+      end
+
+      it 'returns status code 201' do
+        expect(req.stages.first.state).to eq(Stage::CANCELED_STATE)
+        expect(req.stages.last.state).to eq(Stage::SKIPPED_STATE)
+        expect(response).to have_http_status(201)
+      end
+    end
+
+    context 'when request is not actionable' do
+      let(:req) { create(:request, :with_context, :tenant_id => tenant.id) }
```

Comment:
Do we need to create `:req` for each context?

Reply:
Yes, has to link stages to it in each context.